projects/sandbox/src/app/wizard/wizard.component.ts

```diff
@@ -96,18 +96,37 @@ export class WizardComponent extends DataPollingComponent {
   }
 
   /**
-   * validatorWhitespace
+   * validateDatasetId
    *
-   * form validator implementation for whitespace detection
+   * form validator implementation for dataset id field
    *
    * @param { FormControl } control - the control to validate
    * @returns null or a code-keyed boolean
    **/
-  validatorWhitespace(control: FormControl): { [key: string]: boolean } | null {
+  validateDatasetId(control: FormControl): { [key: string]: boolean } | null {
     const val = control.value;
+    const sVal = `${val}`;
     if (val) {
-      const isWhitespace = val.length > 0 && val.trim().length === 0;
-      if (isWhitespace) {
+      if (isNaN(val) || sVal.length === 0 || sVal.length !== `${parseInt(val)}`.length) {
+        return { invalid: true };
+      }
+    }
+    return null;
+  }
+
+  /**
+   * validateDatasetName
+   *
+   * form validator implementation for dataset name field
+   *
+   * @param { FormControl } control - the control to validate
+   * @returns null or a code-keyed boolean
+   **/
+  validateDatasetName(control: FormControl): { [key: string]: boolean } | null {
+    const val = control.value;
+    if (val) {
+      const matches = val.match(/[a-zA-Z0-9\\_\\]+/);
```

Comment:
I think there is something wrong here? This allows backslash twice, right? I think you could just put `/[a-zA-Z0-9_]+/`, right?

Reply:
Yes